nvflare/private/fed/utils/fed_utils.py

```diff
@@ -99,33 +89,141 @@ def listen_command(listen_port, engine, execute_func, logger):
+    """
+    # initialize the SecurityContentService.
+    # must do this before initializing other services since it may be needed by them!
+    startup_dir = workspace.get_startup_kit_dir()
+    SecurityContentService.initialize(content_folder=startup_dir)
+
+    if secure_train:
+        insecure_list = _check_secure_content(site_type=site_type)
+        if len(insecure_list):
+            print("The following files are not secure content.")
+            for item in insecure_list:
+                print(item)
+            sys.exit(1)
+
+    # initialize the AuditService, which is used by command processing.
+    # The Audit Service can be used in other places as well.
+    audit_file_name = workspace.get_audit_file_path()
+    AuditService.initialize(audit_file_name)
+
+    if app_validator:
+        AppAuthzService.initialize(app_validator)
+
+    # Initialize the AuthorizationService. It is used by command authorization
+    # We use FLAuthorizer for policy processing.
+    # AuthorizationService depends on SecurityContentService to read authorization policy file.
+    authorizer = None
+    if secure_train:
+        policy_file_path = workspace.get_authorization_file_path()
+
+        if policy_file_path and os.path.exists(policy_file_path):
+            policy_config = json.load(open(policy_file_path, "rt"))
+            authorizer = FLAuthorizer(site_org, policy_config)
+
+    if not authorizer:
+        authorizer = EmptyAuthorizer()
+
+    _, err = AuthorizationService.initialize(authorizer)
+
+    if err:
+        print("AuthorizationService error: {}".format(err))
+        sys.exit(1)
+
+
+def security_close():
+    AuditService.close()
+
+
+def get_job_meta_from_workspace(workspace: Workspace, job_id: str) -> dict:
+    job_meta_file_path = workspace.get_job_meta_path(job_id)
+    with open(job_meta_file_path) as file:
+        return json.load(file)
+
+
+def create_job_processing_context_properties(workspace: Workspace, job_id: str) -> dict:
+    job_meta = get_job_meta_from_workspace(workspace, job_id)
+    assert isinstance(job_meta, dict), f"job_meta must be dict but got {type(job_meta)}"
+    scope_name = job_meta.get(JobMetaKey.SCOPE, "")
+    scope_object = PrivacyService.get_scope(scope_name)
+    scope_props = None
+    if scope_object:
+        scope_props = scope_object.props
+        effective_scope_name = scope_object.name
+    else:
+        effective_scope_name = ""
+
+    return {
+        FLContextKey.JOB_META: job_meta,
+        FLContextKey.JOB_SCOPE_NAME: scope_name,
+        FLContextKey.EFFECTIVE_JOB_SCOPE_NAME: effective_scope_name,
+        FLContextKey.SCOPE_PROPERTIES: scope_props,
+        FLContextKey.SCOPE_OBJECT: scope_object,
+    }
+
+
+def configure_logging(workspace: Workspace):
+    log_config_file_path = workspace.get_log_config_file_path()
+    assert os.path.isfile(log_config_file_path), f"missing log config file {log_config_file_path}"
```

Comment:
suggest use `if` not `assert` if we want this code to have effect in production code

Reply:
Will address this in another PR